FinSearchUnified/Tests/Bundle/ProductNumberSearchTest.php

```diff
@@ -90,47 +136,58 @@ public function testProductNumberSearchImplementation($isFetchCount, $isUseShopS
         $request = new RequestHttp();
         $request->setModuleName('frontend');
 
-        // Create Mock object for Shopware Front Request
-        $front = $this->getMockBuilder(Front::class)
-            ->setMethods(['Request'])
-            ->disableOriginalConstructor()
-            ->getMock();
-        $front->expects($this->any())
-            ->method('Request')
-            ->willReturn($request);
-
-        // Assign mocked variable to application container
-        Shopware()->Container()->set('front', $front);
+        Shopware()->Front()->setRequest($request);
 
         $context = Shopware()->Container()->get('shopware_storefront.context_service')->getContext();
         $productNumberSearch->search($criteria, $context);
     }
 
-    public function productNumberSearchProvider()
+    /**
+     * @throws Enlight_Exception
+     */
+    public function testFallBackCookieIsSet()
```

Comment:
True, we can not get around the redirection. Let's keep it still. Maybe we know more in the future.